### PR TITLE
Add intra pred unit test

### PR DESCRIPTION
**Pull request by Cidana-Developers**
1. add unit test for all the intra prediction modes except palatte
   and intrabc.
2. add unit test for upsampling and edge filter process.
3. cfl_predict_lbd_avx2 and cfl_predict_hbd_avx2 will produce
    mismatch result compared with c version.

### CODE REVIEW

File: test/intrapred_cfl_test.cc [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+ * Output from assemble functions should be the same with output from c.
+ *
+ * Test coverage:
+ * Test cases:
+ * pred buffer and dst buffer: Fill with random values
+ * TxSize: all the TxSize.
+ * alpha_q3: [-16, 16]
+ * BitDepth: 8bit and 10bit
+ */
+template <typename Pixel, typename FuncType>
```
Comment (andrey-norkin) [inline test/intrapred_cfl_test.cc]: I suggest we use Sample if we refer to a value of a single component, such as a luma sample or a chroma sample. Pixel usually denotes a triplet of values corresponding to a certain spatial position, such as <R,G,B>.

Comment (Cidana-Developers) [reply test/intrapred_cfl_test.cc]: OK, done.
